### PR TITLE
sensors/vehicle_angular_velocity: add new parameter for ESC RPM notchfilter BW

### PR DESCRIPTION
**Pull request by dagar**
- new parameter `IMU_GYRO_DNF_BW` for setting the bandwidth of the ESC RPM notch filters
 - existing `IMU_GYRO_DYN_NF` parameter was renamed `IMU_GYRO_DNF_EN` for consistency
 - new perf counters for FFT and ESC RPM dynamic notch filtering when the filters are reset or fully disabled
    - this is mainly for monitoring things at this stage and can probably be removed eventually

Here's current master on a quad with and without ESC RPM notch filtering enabled.
Screenshot from 2021-11-03 14-47-52

### CODE REVIEW

**Review by AlexKlimaj — DISMISSED** [review]

**Comment by hamishwillee** [comment]
@dagar @AlexKlimaj @bresch Can you guys commit to some docs when this goes in or suggest who might help with this? This would be an update to at least this doc http://docs.px4.io/master/en/config_mc/filter_tuning.html

The update would cover:
- What the filter is in http://docs.px4.io/master/en/config_mc/filter_tuning.html#filters - i.e. an update or replacement to:
  > A notch filter on the gyro sensor data that is used to filter out narrow band noise, for example harmonics at the rotor blade pass frequency. This filter can be configured using IMU_GYRO_NF_BW and IMU_GYRO_NF_FREQ.
- Update to [Filter tuning section](http://docs.px4.io/master/en/config_mc/filter_tuning.html#filter-tuning) to explain how you set this based on logs. 

Further, Ramon will probably want to blog about this and would be interested in the PR as a "change" - ie. how does this improve previous behaviour.

@bresch , specifically to you, I'd hope this would result in an update to http://docs.px4.io/master/en/flight_stack/controller_diagrams.html#multicopter-angular-rate-controller

It would be great if we could do this before the next PX4 release comes out :-)

File: src/lib/mathlib/math/filter/LowPassFilter2p.hpp [hunk 1]
```diff
@@ -71,7 +71,7 @@ class LowPassFilter2p
 		_delay_element_1 = {};
 		_delay_element_2 = {};
 
-		_cutoff_freq = math::constrain(cutoff_freq, 5.f, sample_freq / 2); // TODO: min based on actual numerical limit
+		_cutoff_freq = math::max(cutoff_freq, sample_freq * 0.001f);
```
Comment (bresch) [inline src/lib/mathlib/math/filter/LowPassFilter2p.hpp:74]: Why did you remove the upper bound? That one can still be the Nyquist frequency or a fraction of it.

Comment (dagar) [reply src/lib/mathlib/math/filter/LowPassFilter2p.hpp:74]: It's already handled above (disabling the filter entirely). https://github.com/PX4/PX4-Autopilot/blob/acf51d54b5d33ab65e78810ad46f3121274d6346/src/lib/mathlib/math/filter/LowPassFilter2p.hpp#L63-L68

**Review by bresch — APPROVED** [review]
